src/classes/ERR_RecordError.cls

```diff
@@ -0,0 +1,536 @@
+    * @param ts The timestamp the associated
+    * @param context The context the exception was raised on.
+    * @param objectType The type of all records in the job, only to be passed in if they all match.
+    * @return The error record.
+    */
+    @testVisible private static String updateFullMessage(Datetime ts, String existingMessage, String message) {
+
+        String updatedMessage = ts.format() + ':\r\n' + message;
+
+        if (String.isNotBlank(existingMessage)) {
+            updatedMessage += '\r\n' + '========================================\r\n' + existingMessage;
+            // Limit to 5K worth of Message History
+            updatedMessage = updatedMessage.abbreviate(5120);
+        }
+
+        return updatedMessage;
+    }
+
+    /*******************************************************************************************************
+    * @description Returns the URL of a record, if available.
+    * @param recordId The id of the record associated with the error
+    * @return The URL of the record. Warning: it may not necessarily be valid.
+    */
+    private static String getRecordURL(String recordId) {
+        String recordURL = URL.getSalesforceBaseUrl().toExternalForm() + '/' + recordId;
+
+        recordURL = recordURL.replaceFirst('http:', 'https:');
+
+        return recordURL;
+    }
+
+    /*******************************************************************************************************
+    * @description Returns the status of the error that happened as a result of a DML operation performed through
+    *              the Database class as a string.  If null, will return null
+    * @param dmlError The DML error that occurred
+    * @return The error message as a String
+    */
+    private static String getDmlMessage(Database.Error dmlError) {
+        return (dmlError != null) ? dmlError.getMessage() : null;
+    }
+
+    /*******************************************************************************************************
+    * @description Returns the status of the error that happened as a result of a DML operation performed through
+    *              the Database class as a string.  If null, will return null
+    * @param dmlError The DML error that occurred
+    * @return The error status as a string.
+    */
+    private static String getStatusCode(Database.Error dmlError) {
+        return (dmlError != null) ? getStatusCode(dmlError.getStatusCode()) : '(Error not available.)';
+    }
+
+
+    /*******************************************************************************************************
+    * @description Returns the status of the error that happened as a result of a DML operation performed through
+    *              the Database class as a string.
+    * @param status The status code.
+    * @return The error status as a string.
+    */
+    private static String getStatusCode(System.StatusCode status) {
```

Comment:
Isn't this also the same as the version in `ERR_Handler`?

Reply:
If you are ok with changing the visibility on the original, I will do so.

Reply:
I think in this case it might be worth making it public.